bigquery_etl/deploy.py

```diff
@@ -40,12 +45,24 @@ def deploy_table(
     if respect_dryrun_skip and str(artifact_file) in DryRun.skipped_files():
         raise SkippedDeployException(f"Dry run skipped for {artifact_file}.")
 
+    metadata = None
     try:
         if artifact_file.name == METADATA_FILE:
             metadata = Metadata.from_file(artifact_file)
         else:
             metadata = Metadata.of_query_file(artifact_file)
 
+        if metadata.external_data:
+            if str(artifact_file).endswith("query.sql"):
```

Comment:
Should we check for other types of query files, like `query.py` here as well. Maybe similar to https://github.com/mozilla/bigquery-etl/blob/5d216fa2dcbb5674f1e42ad4ebeaf7a5369edefa/bigquery_etl/cli/query.py#L2222-L2226